robot-core/src/main/java/org/obolibrary/robot/template/Template.java

```diff
@@ -0,0 +1,1852 @@
+  private List<List<String>> tableRows;
+
+  /** Row number tracker. */
+  private int rowNum = 0;
+
+  /** Shared data factory. */
+  private final OWLDataFactory dataFactory = OWLManager.getOWLDataFactory();
+
+  /** Namespace for error messages. */
+  private static final String NS = "template#";
+
+  /**
+   * Error message when the number of header columns does not match the number of template columns.
+   * Expects: table name, header count, template count.
+   */
+  private static final String columnMismatchError =
+      NS
+          + "COLUMN MISMATCH ERROR the number of header columns (%2$d) must match "
+          + "the number of template columns (%3$d) "
+          + "in table \"%1$s\".";
+
+  /** Error message when an invalid individual type is provided. */
+  private static final String individualTypeError =
+      NS + "INDIVIDUAL TYPE ERROR individual %s has unknown type '%s' at row %d, column %d";
+
+  /**
+   * Error message when a template cannot be understood. Expects: table name, column number, column
+   * name, template.
+   */
+  private static final String unknownTemplateError =
+      NS
+          + "UNKNOWN TEMPLATE ERROR could not interpret template string \"%4$s\" "
+          + "for column %2$d (\"%3$s\") "
+          + "in table \"%1$s\".";
+
+  /**
+   * Given a template name and a list of rows, create a template object with a new IOHelper and
+   * QuotedEntityChecker. The rows are added to the object, new labels from the rows are added to
+   * the checker, and a Manchester Syntax parser is created.
+   *
+   * @param name template name
+   * @param rows list of rows (lists)
+   * @throws Exception on issue creating IOHelper or adding table to template object
+   */
+  public Template(String name, List<List<String>> rows) throws Exception {
+    this.ioHelper = new IOHelper();
+
+    checker = new QuotedEntityChecker();
+    checker.setIOHelper(this.ioHelper);
+    checker.addProvider(new SimpleShortFormProvider());
+    checker.addProperty(dataFactory.getRDFSLabel());
+
+    tableRows = new ArrayList<>();
+    templates = new ArrayList<>();
+    headers = new ArrayList<>();
+    axioms = new HashSet<>();
+
+    // Add the contents of the tableRows
+    addTable(name, rows);
+
+    // Set all the new labels used in the template
+    // Other labels can be found from the input ontology with the checker
+    addLabels();
+    createParser();
+  }
+
+  /**
+   * Given a template name, a list of rows, and an IOHelper, create a template object with a new
+   * QuotedEntityChecker. The rows are added to the object, new labels from the rows are added to
+   * the checker, and a Manchester Syntax parser is created.
+   *
+   * @param name template name
+   * @param rows list of rows (lists)
+   * @param ioHelper IOHelper to resolve prefixes
+   * @throws Exception on issue adding table to template object
+   */
+  public Template(String name, List<List<String>> rows, IOHelper ioHelper) throws Exception {
```

Comment:
This constructor is almost identical to the previous one. I wish the code could be more DRY, but maybe that's just Java.